### PR TITLE
Allow ttl field in notification

### PR DESCRIPTION
**Pull request by kanes115**

### CODE REVIEW

File: lib/pigeon/notification.ex [hunk 1]
```diff
@@ -87,6 +89,9 @@ defmodule Pigeon.GCM.Notification do
   def put_priority(n, nil),       do: n
   def put_priority(n, priority),  do: update_payload(n, "priority", priority)
 
+  def put_ttl(n, nil),       do: n
```
Comment (rslota) [inline lib/pigeon/notification.ex]: Something is wrong with formatting here and line below (spaces before `do`).

File: lib/pigeon/notification.ex [hunk 2]
```diff
@@ -52,6 +52,8 @@ defmodule Pigeon.APNS.Notification do
 
   def put_priority(notification, priority), do: %{notification | priority: priority}
 
+  def put_ttl(notification, ttl), do: %{notification | time_to_live: ttl}
```
Comment (rslota) [inline lib/pigeon/notification.ex]: APNS doesn't seem to support this at all. Can you share here a link to documentation where you found it? If you haven't, please confirm that it's indeed not possible to get this behavior with APNS ale let's drop it here - there's no point in sending something that will be simply ignored, not to mention that is misleading.

**Review by rslota — CHANGES_REQUESTED** [review]

File: lib/pigeon/notification.ex [hunk 3]
```diff
@@ -71,7 +71,7 @@ defmodule Pigeon.GCM.Notification do
   @moduledoc """
     Defines GCM notification struct and convenience constructor functions.
   """
-  defstruct registration_id: nil, payload: %{}, message_id: nil, updated_registration_id: nil
+  defstruct registration_id: nil, payload: %{}, message_id: nil, updated_registration_id: nil, time_to_live: nil
```
Comment (rslota) [inline lib/pigeon/notification.ex]: Do we really need this? `"time_to_live"` seems to be updated within `payload`.

Comment (kanes115) [reply lib/pigeon/notification.ex]: Of course, yes

**Review by rslota — APPROVED** [review]

**Pull request merged**